lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -350,114 +356,71 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
     config_ = uc.updated_config;
   }
 
+  // Rule versions
+  if (uc.policy_version_and_stats) {
+    policy_version_and_stats_ = *uc.policy_version_and_stats;
+  }
+
   // Static rules
   for (const auto& rule_id : uc.static_rules_to_uninstall) {
     if (is_static_rule_installed(rule_id)) {
-      deactivate_static_rule(rule_id, _);
-    } else if (is_static_rule_scheduled(rule_id)) {
-      install_scheduled_static_rule(rule_id, _);
-      deactivate_static_rule(rule_id, _);
-    } else {
-      MLOG(MERROR) << "Failed to merge: " << session_id_
-                   << " because static rule already uninstalled: " << rule_id;
-      return false;
+      remove_from_vec_by_value<std::string>(active_static_rules_, rule_id);
```

Comment:
i read the summary, but why is the reason we can't use deactivate herer?

Reply:
Actually will update the next PR with documentation, since this PR does not include the version update change yet :)